pkg/diagnostics/factory.go

```diff
@@ -37,17 +37,17 @@ func NewFactory(opts EksaDiagnosticBundleFactoryOpts) *eksaDiagnosticBundleFacto
 
 func (f *eksaDiagnosticBundleFactory) DiagnosticBundle(spec *cluster.Spec, provider providers.Provider, kubeconfig string, bundlePath string) (DiagnosticBundle, error) {
 	if bundlePath == "" && spec != nil {
-		b, err := f.DiagnosticBundleFromSpec(spec, provider, kubeconfig)
+		b, err := f.DiagnosticBundleWorkloadCluster(spec, provider, kubeconfig)
 		return b, err
 	}
 	return f.DiagnosticBundleCustom(kubeconfig, bundlePath), nil
 }
 
-func (f *eksaDiagnosticBundleFactory) DiagnosticBundleManagementCluster(kubeconfig string) (DiagnosticBundle, error) {
-	return newDiagnosticBundleManagementCluster(f.analyzerFactory, f.collectorFactory, f.client, f.kubectl, kubeconfig, f.writer)
+func (f *eksaDiagnosticBundleFactory) DiagnosticBundleManagementCluster(spec *cluster.Spec, kubeconfig string) (DiagnosticBundle, error) {
```

Comment:
If I understand this correctly, we might want to rename it to `DiagnosticBundleBootstrapCluster`. Looks like this method fetches kind/bootstrap cluster logs before the CAPI move.

Reply:
This method gets invoked every time the `CollectDiagnosticsTask` task is run in workflows/diagnostics.go. If we're performing an operation from a management cluster to a workload cluster and something goes wrong and we start collecting diagnostics, it'll collect them from the management cluster, since there's no bootstrap cluster. So it seems like the terminology still doesn't make sense in that scenario. I can't think of the right thing to call this method since the definition of "management cluster" seems to be fluid. If a bootstrap cluster is in use, then that's the current management cluster. 

Is there a word for the cluster that's orchestrating the CLI operation?

Reply:
I dont know that.. I've been confused by those terms in our codebase. I think we have a project to clean it up bit. https://github.com/aws/eks-anywhere/projects/10

Reply:
I'm looking forward to re-learning the eks-a code flow through this lens